lib/services/Task/methods/delete.ts

```diff
@@ -0,0 +1,58 @@
+import { db } from '@/lib/db';
+import type { ServiceResponse } from '@/lib/services';
+
+export async function deleteTask(
+  id: string
+): Promise<ServiceResponse<boolean>> {
+  try {
```

Comment:
We can take the time to add that nuanced behavior to delete - check if `syncStatus` is `new`.  That means it hasn't ever been sent to AWS, that database has no idea about it and never needs to. We can remove it from the db right here.